### DISCUSSION TITLE
How to install ROCm 6.2 on Pop! OS?

### DISCUSSION THREAD

**Discussion by jessecambon** [ROCm/ROCm#3907] [Q&A] [answered]
I have been able to successfully install and use ROCm 6.1 on Pop OS! using [these instructions](https://support.system76.com/articles/rocm/) and the ROCm docker images from AMD. Because of the improvements in memory utilization and mixed precision, I'm looking to upgrade to ROCm 6.2. 

I've attempted to install 6.2 by pointing the apt entry in the file `/etc/apt/sources.list.d/rocm.list` to `https://repo.radeon.com/rocm/apt/6.2.2` (instead of 6.1). This results in the following error when attempting to install the `rocm` apt package:

```
The following packages have unmet dependencies:
 rocdecode : Depends: mesa-amdgpu-va-drivers but it is not installable
E: Unable to correct problems, you have held broken packages.
```

What should I try next here?  For reference I have a 7700 XT GPU, Ryzen 7600 CPU, and the Linux kernel is `6.9.3-76060903-generic`.

**Comment by darren-amd** [comment]
Hi @jessecambon,

I was able to find a similar issue [here](https://www.reddit.com/r/pop_os/comments/1f384kb/pop_os_2404_lts_rocm_and_others/?rdt=42945) from another user running on PopOS. Please make sure to follow the official [install instructions](https://rocm.docs.amd.com/projects/install-on-linux/en/latest/install/quick-start.html) and specifically make sure to install the amdgpu package. The missing `mesa-amdgpu-va-drivers` are currently packaged within this package rather than rocm, resulting in the error you are facing. Please run the following install script and let me know if you run into any issues:

```
sudo apt update
sudo apt install "linux-headers-$(uname -r)" "linux-modules-extra-$(uname -r)"
sudo usermod -a -G render,video $LOGNAME # Add the current user to the render and video groups
wget https://repo.radeon.com/amdgpu-install/6.2.4/ubuntu/noble/amdgpu-install_6.2.60204-1_all.deb
sudo apt install ./amdgpu-install_6.2.60204-1_all.deb
sudo apt update
sudo apt install amdgpu-dkms rocm
```

**Reply by jessecambon** [reply]
Hi @darren-amd, I get the following error on the second command. I believe this has to do with the linux kernel that is installed in Pop OS!

```sh
$ sudo apt install "linux-headers-$(uname -r)" "linux-modules-extra-$(uname -r)"
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Package linux-modules-extra-6.9.3-76060903-generic is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'linux-modules-extra-6.9.3-76060903-generic' has no installation candidate
```

**Comment by new-php** [comment, marked as answer]
<p>I was struggling to install ROCm 6.2 on Pop! OS, but after following a few steps, I managed to get it up and running. Here's what worked for me:</p>

  <ol>
    <li><strong>Add ROCm Repository</strong><br>First, I added the ROCm repository to my system by running the following commands:
      <pre><code>
sudo apt update
sudo apt install wget
wget -qO - https://packages.amd.com/rocm/rocm.gpg | sudo tee /etc/apt/trusted.gpg.d/rocm.asc
sudo sh -c 'echo "deb [arch=amd64] https://packages.amd.com/rocm/apt/debian/ rocm main" > /etc/apt/sources.list.d/rocm.list'
      </code></pre>
    </li>

    <li><strong>Install ROCm 6.2</strong><br>After the repository was added, I ran the following commands to install ROCm:
      <pre><code>
sudo apt update
sudo apt install rocm-dkms
      </code></pre>
    </li>

    <li><strong>Set Up Permissions for the User</strong><br>To ensure that I have the right permissions, I added my user to the video group:
      <pre><code>
sudo usermod -aG video $USER
      </code></pre>
    </li>

    <li><strong>Reboot the System</strong><br>I rebooted the system for the changes to take effect:
      <pre><code>
sudo reboot
      </code></pre>
    </li>

    <li><strong>Verify Installation</strong><br>Finally, I verified the installation using:
      <pre><code>
/opt/rocm/bin/rocminfo
      </code></pre>
      This command should show detailed information about my ROCm setup.
    </li>
  </ol>

  <p>If you're facing issues with this, I highly recommend referring to the <a href="https://docs.vultr.com/" target="_blank"><strong>Vultr Guide</strong></a>. It helped me navigate through the setup process and resolve network-related issues when running ROCm on a cloud instance. Additionally, if you're running this setup on a server instance (like on <strong>Vultr</strong>), checking their documentation for configuring networking properly can save a lot of time and prevent issues.</p>

  <p>Now everything is working fine for me with ROCm 6.2 on Pop! OS!</p>
</div>

**Reply by jessecambon** [reply]
Hi @new-php , thanks for writing this up. Are you still using this same repository? I tried following along, but I'm not seeing anything at https://packages.amd.com/.  If I manually add it to my sources I get `Could not connect to packages.amd.com:443 (165.204.85.111), connection timed out`
`
